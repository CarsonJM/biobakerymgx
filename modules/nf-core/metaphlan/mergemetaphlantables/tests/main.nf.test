nextflow_process {

    name "Test Process: METAPHLAN_MERGEMETAPHLANTABLES"
    script "../main.nf"
    process "METAPHLAN_MERGEMETAPHLANTABLES"

    tag "modules"
    tag "modules_local"
    tag "metaphlan"
    tag "metaphlan/mergemetaphlantables"


    test(" ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] ") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file( params.test_data['assets']['metaphlan']['metaphlan_sgb_profile1_txt'], checkifExists: true)
                    ],
                    [
                        [ id: 'test2' ],
                        file( params.test_data['assets']['metaphlan']['metaphlan_sgb_profile2_txt'], checkifExists: true)
                    ]
                )
                .map { [ [ id:'all_samples' ], it[1] ] }
                .groupTuple( sort: 'deep' )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test(" ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] -stub ") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file( params.test_data['assets']['metaphlan']['metaphlan_sgb_profile1_txt'], checkifExists: true)
                    ],
                    [
                        [ id: 'test2' ],
                        file( params.test_data['assets']['metaphlan']['metaphlan_sgb_profile2_txt'], checkifExists: true)
                    ]
                )
                .map { [ [ id:'all_samples' ], it[1] ] }
                .groupTuple( sort: 'deep' )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
