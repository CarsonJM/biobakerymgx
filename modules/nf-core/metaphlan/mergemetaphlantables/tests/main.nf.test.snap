{
    " ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,f4627b6001fe6dbde372f0a648f574c2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,36d4842de5dd44b8b642cbfbbc6b4aff"
                ],
                "txt": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,f4627b6001fe6dbde372f0a648f574c2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,36d4842de5dd44b8b642cbfbbc6b4aff"
                ]
            }
        ],
        "timestamp": "2024-01-26T08:14:43.460606117"
    },
    " ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] -stub ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,36d4842de5dd44b8b642cbfbbc6b4aff"
                ],
                "txt": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,36d4842de5dd44b8b642cbfbbc6b4aff"
                ]
            }
        ],
        "timestamp": "2024-01-26T08:14:56.286600733"
    }
}